eventcontent.go

```diff
@@ -137,6 +148,37 @@ func newMemberContentFromEvent(event Event) (c memberContent, err error) {
 	return
 }
 
+// thirdPartyInviteContent is the JSON content of a m.room.third_party_invite event needed for auth checks.
+// See https://matrix.org/docs/spec/client_server/r0.2.0.html#m-room-third-party-invite for descriptions of the fields.
+type thirdPartyInviteContent struct {
+	// Public keys are used to verify the signature of a m.room.member event that
+	// came from a m.room.third_party_invite event
+	PublicKeys []struct {
+		PublicKey Base64String `json:"public_key"`
+	} `json:"public_keys"`
+}
+
+func newThirdPartyInviteContentFromAuthEvents(authEvents AuthEventProvider, token string) (t thirdPartyInviteContent, err error) {
+	var thirdPartyInviteEvent *Event
+	if thirdPartyInviteEvent, err = authEvents.ThirdPartyInvite(token); err != nil {
+		return
+	}
+	if thirdPartyInviteEvent == nil {
+		// If there isn't a third_party_invite event, then we return with an error
+		err = errorf("Couldn't find third party invite event")
+		return
+	}
+	return newThirdPartyInviteContentFromEvent(*thirdPartyInviteEvent)
+}
+
+func newThirdPartyInviteContentFromEvent(event Event) (t thirdPartyInviteContent, err error) {
+	if err = json.Unmarshal(event.Content(), &t); err != nil {
+		err = errorf("unparsable third party invite event content: %s", err.Error())
+		return
+	}
+	return
+}
```

Comment:
The `newThirdPartyInviteContentFromEvent` function doesn't seem to be doing much here, would it make sense to merge it into the `newThirdPartyInviteContentFromAuthEvents` function above?

Reply:
Done https://github.com/matrix-org/gomatrixserverlib/pull/58/commits/d05d616c70ca8d2e6daac1b487f710ebc2dbf090